transport/src/main/java/io/netty/channel/CoalescingBufferQueue.java

```diff
@@ -37,11 +37,16 @@
 public final class CoalescingBufferQueue {
 
     private final Channel channel;
-    private final ArrayDeque<Object> bufAndListenerPairs = new ArrayDeque<Object>();
+    private final ArrayDeque<Object> bufAndListenerPairs;
     private int readableBytes;
 
     public CoalescingBufferQueue(Channel channel) {
+        this(channel, 16);
```

Comment:
+1. I'm fine with going lower, and setting a lower default ... lets if @louiscryan has any concerns.